src/graphql/types/object/transaction.ts

```diff
@@ -81,21 +84,29 @@ const Transaction = GT.Object<WalletTransaction>({
       type: GT.NonNull(SignedAmount),
     },
     settlementPrice: {
-      type: GT.NonNull(Price),
+      type: GT.NonNull(SettlementPriceInMinorUnit),
       resolve: (source) => {
-        const displayCurrencyPerSettlementCurrencyUnitInCents =
-          source.displayCurrencyPerSettlementCurrencyUnit * 100
+        const displayCurrency = checkedToDisplayCurrency(source.settlementDisplayCurrency)
+        if (displayCurrency instanceof Error) throw mapError(displayCurrency)
+
+        const displayCurrencyPriceInMinorUnit = currencyMajorToMinorUnit({
+          amount: source.displayCurrencyPerSettlementCurrencyUnit,
+          displayCurrency,
+        })
+
+        let offset = SAT_PRICE_PRECISION_OFFSET
+        if (source.settlementCurrency === WalletCurrencyDomain.Usd) {
+          offset = USD_PRICE_PRECISION_OFFSET
+        }
+
         return {
-          formattedAmount: displayCurrencyPerSettlementCurrencyUnitInCents.toString(),
-          base: Math.round(
-            displayCurrencyPerSettlementCurrencyUnitInCents *
-              10 ** SAT_PRICE_PRECISION_OFFSET,
-          ),
-          offset: SAT_PRICE_PRECISION_OFFSET,
-          currencyUnit: "USDCENT",
+          base: Math.round(displayCurrencyPriceInMinorUnit * 10 ** offset),
+          offset,
+          currencyUnit: "MINOR",
+          formattedAmount: displayCurrencyPriceInMinorUnit.toString(),
```

Comment:
Would `toString()` always work, or should we do something like `${displayCurrencyPriceInMinorUnit}`

Reply:
Updated https://github.com/GaloyMoney/galoy/pull/2439/commits/1dfb099095a98c412a0e220e2342ab0ccf04b6a3